src/components/Card.js

```diff
@@ -9,9 +12,17 @@
       this._cardElement.querySelector('.card__link').addEventListener('click', () => {
         this._handleLikeButton();
       });
+      
       this._cardElement.querySelector('.card__delete-btn').addEventListener('click', () => {
-        this._handleDeleteCard();
+        const deleteCardPopup = createPopupWithYes('#delete-card-modal', () => {
+          api.deleteCard(this._data._id);
+          this._handleDeleteCard();
+        });
+  
+        deleteCardPopup.open();
+        deleteCardPopup.setEventListeners();
       });
+      
       this._cardElement.querySelector('.card__image').addEventListener('click', () => {
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.

```js
  returnCardElement() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```